pkg/runtime/registry/options.go

```diff
@@ -0,0 +1,122 @@
+/*
+Copyright 2021 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package registry
+
+import (
+	"github.com/dapr/dapr/pkg/components/bindings"
+	"github.com/dapr/dapr/pkg/components/configuration"
+	"github.com/dapr/dapr/pkg/components/crypto"
+	"github.com/dapr/dapr/pkg/components/lock"
+	"github.com/dapr/dapr/pkg/components/middleware/http"
+	"github.com/dapr/dapr/pkg/components/nameresolution"
+	"github.com/dapr/dapr/pkg/components/pubsub"
+	"github.com/dapr/dapr/pkg/components/secretstores"
+	"github.com/dapr/dapr/pkg/components/state"
+	"github.com/dapr/dapr/pkg/components/workflows"
+)
+
+// Options is the options to configure the registries
+type Options struct {
```

Comment:
The `components` may be more meaningful.

Reply:
>  The components may be more meaningful.

I don't know what this means.

> I prefer to NewRegistry register all default registry, and WithXXX in Registry, no need to create Options.

Options are very important here for ensuring that the registry pointers cannot be changed after `NewRegistry` is called, i.e. changing a registry store mid lifecycle of a runtime.